mars/core/operand/base.py

```diff
@@ -142,6 +142,12 @@ class Operand(Base, metaclass=OperandMetaclass):
     _inputs = ListField("inputs", FieldTypes.reference(EntityData))
     _outputs = ListField("outputs")
     _output_types = ListField("output_type", FieldTypes.reference(OutputType))
+    # An unique and deterministic key for subtask compute logic. It should be even the same
+    # for different run if the compute logic doesn't change. This id will be used in subtask speculative
+    # execution and hbo scheduling and so on.
+    logic_key = StringField('logic_key')
```

Comment:
Speculation is part of execution strategy, and it is better to separate them from core data structure.

Reply:
It's also used in hbo(history based optimization) for match history subtasks.